x/iro/keeper/params.go

```diff
@@ -23,6 +25,47 @@ func (m msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParam
 	if err != nil {
 		return nil, err
 	}
+	/* -------------------------------------------------------------------------- */
+	/*                 stateful validation for fair launch params                 */
+	/* -------------------------------------------------------------------------- */
+	fairLaunch := req.NewParams.FairLaunch
+
+	// validate target raise denom is allowed
+	targetRaise := fairLaunch.TargetRaise
+	targetRaiseMetadata, ok := m.BK.GetDenomMetaData(ctx, targetRaise.Denom)
+	if !ok {
+		return nil, errorsmod.Wrapf(gerrc.ErrInvalidArgument, "denom %s not registered", targetRaise.Denom)
+	}
+	targetRaiseExponent := targetRaiseMetadata.DenomUnits[len(targetRaiseMetadata.DenomUnits)-1].Exponent
```

Comment:
same as my previos comment about being sure it's always the last in the list which has the biggest decimal